crates/storage/provider/src/providers/static_file/mod.rs

```diff
@@ -150,7 +150,9 @@ mod tests {
         // Use providers to query Header data and compare if it matches
         {
             let db_provider = factory.provider().unwrap();
-            let manager = StaticFileProvider::new(static_files_path.path()).unwrap().with_filters();
+            let manager = StaticFileProvider::new(static_files_path.path(), StaticFileEnv::RW)
```

Comment:
can we add `fn read_write` and `fn read_only` that calls new with the corresponding enum for convenience?